File: back-end/src/models/Linha.ts
/* eslint-disable camelcase */
import {
  Entity, Column, PrimaryGeneratedColumn, CreateDateColumn, UpdateDateColumn,
  ManyToMany, JoinTable, OneToMany,
} from 'typeorm';

import Parada from './Parada';
import Veiculo from './Veiculo';

  @Entity('Linha')
class Linha {
      @PrimaryGeneratedColumn('increment')
      Id: bigint;

      @Column('varchar')
      Name: string;

      @ManyToMany(() => Parada, (Paradas:Parada) => Paradas.Linhas)
      @JoinTable()
      paradas: Parada[];

      @OneToMany(() => Veiculo, (veiculo: Veiculo) => veiculo.linhaId)
      Veiculos: Veiculo[];

      @CreateDateColumn()
      created_at: Date;

      @UpdateDateColumn()
      updated_at: Date;
  }

export default Linha;

File: Models/Linha.cs
using System;
using System.Collections;
using programa_estagio_2021.Models;

namespace programa_estagio_2021.Models
{
    class Linha 
    {
        public long Id { get; set;}
        public string Name {get; set;}
        public int[] Paradas = new int[5];

        public Linha()
        {
        }
    }
}